### ISSUE TITLE
queryRenderedFeaturesAtPoint returns no custom properties with mapbox-android-sdk:9.4.0+ and Mapbox-iOS-SDK 6.1.0+

### ISSUE THREAD

**Issue by hexadecy** [rnmapbox/maps#1402]
**Describe the bug**  
queryRenderedFeaturesAtPoint returns empty feature.properties with mapbox-android-sdk:9.4.0+ and Mapbox-iOS-SDK 6.1.0+

**To Reproduce**  
1. Provides Mapbox style from a custom private StyleURL which contains:
```json
"layers": [
  {
    "id": "ndvi-date-layer",
    "type": "fill",
    "source": "ndvi-date-source",
    "visibility": "none"
   },
 ],
 "sources": {
    "composite": {
      "type": "vector",
      "url": "mapbox://mapbox.mapbox-streets-v8"
    },
    "ndvi-date-source": {
      "type": "geojson",
      "data": {
        "type": "FeatureCollection",
        "features": [
          {
            "type": "Feature",
            "geometry": {
              "type": "Polygon",
              "coordinates": [
                [
                  [
                    -73.2031253995361,
                    0.8154182713648
                  ],
                  [
                    -72.21601710588774,
                    0.8154182713648
                  ],
                  [
                    -72.21601710588774,
                    1.80892254621211
                  ],
                  [
                    -73.2031253995361,
                    1.80892254621211
                  ],
                  [
                    -73.2031253995361,
                    0.8154182713648
                  ]
                ]
              ]
            },
            "properties": {
              "dataset-id": "18NYG",
              "timestamp": "2020-02-15T15:22:34.899Z"
            }
          },
        ],
      }
    }
  }
```

2. Get the features from the center of the map
```js
const center = await mapRef.current.getCenter();
const pointInView = await mapRef.current.getPointInView(center);
if (Platform.OS == 'android') {
  pointInView[0] *= PixelRatio.get();
  pointInView[1] *= PixelRatio.get();
}
const { features } = await mapRef.current.queryRenderedFeaturesAtPoint(pointInView, ['==', '$type', 'Polygon']);
const [feature] = features;
const { timestamp } = feature.properties;
```
**Expected behavior**  
timestamp shoud be "2021-06-08T19:03:41.186Z"

**Actual behavior**  
dataset-id and timestamp are not found in feature.properties

**Versions:**  
 - Platform: [Android, iOS]
 - Platform OS: [Android 9, iOS 14.5]
 - Device: [Samsung S8, iPhone 11]
 - Emulator/ Simulator: [same issue]
 - Dev OS: [OSX BigSur]
 - react-native-mapbox-gl Version [8.2.0-beta2]
 - Mapbox-iOS-SDK [6.1.0-6.3.0]
 - mapbox-android-sdk [9.4.0-9.6.1]
 - React Native Version [0.63.4]

**Additional info**
It was working with mapbox-android-sdk 9.3.1 and Mapbox-iOS-SDK 6.0.0 and below.

**Comment by liuminghui2017** [comment]
try this on Android:
`
queryRenderedFeaturesAtPoint([x * PixelRatio.get(), y * PixelRatio.get()])
`

**Comment by hexadecy** [comment]
Yes I saw that in https://github.com/react-native-mapbox-gl/maps/issues/1085.
I tried it and it changes the position not that much. 
I'm using sentinel 2 tiles which are 100km x 100km.

I suspect a bump in the mapbox core version:
Update mobile-maps-gl-core library from 1.8.1 to 3.1.0

**Comment by hexadecy** [comment]
Maybe it was a bug fix in mobile-maps-gl-core 3.1.0 that broke it.
```
"layers": [
  {
    "id": "ndvi-date-layer",
    "type": "fill",
    "source": "ndvi-date-source",
    "visibility": "none"
  }
]
```
With newer versions the layer must be VISIBLE.
```
"layers": [
  {
    "id": "ndvi-date-layer",
    "type": "fill",
    "source": "ndvi-date-source",
    "paint": {
      "fill-opacity": 0
    }
  }
]
```
